ehrapy/api/preprocessing/normalization.py

```diff
@@ -0,0 +1,109 @@
+from typing import Dict, Optional, Union
+
+from anndata import AnnData
+from numpy import array
+from sklearn.preprocessing import minmax_scale, scale
+
+from ehrapy.api._anndata_util import assert_encoded, get_column_indices, get_column_values, get_numeric_vars
+
+
+class Normalization:
+    """Provides functions to normalize continuous features"""
+
+    available_methods = {"scale", "minmax", "identity"}
+
+    @staticmethod
+    def _normalize(adata: AnnData, methods: Union[Dict[str, str], str], copy: bool = False) -> Optional[AnnData]:
+        """Normalize numeric variable.
+
+        This function normalizes the numeric variables in an AnnData object.
+
+        Available normalization methods are:
+
+        1. scale (https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.scale.html#sklearn.preprocessing.scale)
+        2. minmax (https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MinMaxScaler.html#sklearn.preprocessing.MinMaxScaler)
+        3. identity (return the un-normalized values)
+
+        Args:
+            adata: :class:`~anndata.AnnData` object containing X to normalize values in. Must already be encode using ~ehrapy.preprocessing.encode.encode.
+            methods: Methods to use for normalization. Either:
+
+                str: Name of the method to use for all numeric variable
+
+                Dict: A dictionary specifying the method for each numeric variable where keys are variable and values are methods
+            copy: Whether to return a copy or act in place
+
+        Returns:
+            :class:`~anndata.AnnData` object with normalized X
+        """
+
+        assert_encoded(adata)
+
+        num_vars = get_numeric_vars(adata)
+        if isinstance(methods, str):
+            methods = dict.fromkeys(num_vars, methods)
+        else:
+            if not set(methods.keys()) <= set(num_vars):
+                raise ValueError("Some keys of methods are not numeric variables")
```

Comment:
Will eventually also be solved with https://github.com/theislab/ehrapy/issues/182
But again, let 's keep it here.